R/00.salmonag.R

```diff
@@ -496,11 +698,49 @@ plot(clip2)
+# This is also used as the substitution matrix to group all land use types into crop categories
+types<- df_clip2  %>%
+  group_by(cdl_west)  %>%
+  count
+
+types<-types %>%
+  mutate(cropgroup=if_else(str_detect(cdl_west, "Barley|Rye|Wheat|Oats|Triticale|Sorghum|Buckwheat|Millet|Dbl"), "smgrains", 
+                           if_else(str_detect(cdl_west, "Walnuts|Almonds|Pecans|Pistachios"), "nuts" ,
+                                   if_else(str_detect(cdl_west, "Apples  | Apricots|berries|  Cantaloupes|Cherries|Citrus|Watermelons|Melons|Nectarines|Oranges|Peaches|Pears|Plums|Pomegranates|Other Tree|Olives"), "fruits",
+                                           if_else(str_detect(cdl_west, "Tomatoes|Asparagus|Broccoli|Cabbage|Carrots|Cauliflower|Cucumbers|Garlic|Gourds|Greens|Lettuce|Onions|Peas|Peppers|Popcorn|Pumpkins|Radish|Squash|Sweet Corn|Sweet Potatoes|Turnips"), "veg",
+                                                   if_else(str_detect(cdl_west,  "Chick Peas|Christmas Trees|Dry Beans|Flaxseed|Herbs|Hops|Lentils|Mint|Misc Vegs|Mustard|Other Crops|Rape|Safflower|Sod/Grass Seed|Sugarbeets|Sunflower|Vetch"), "other_crops",                
+                                                           if_else(str_detect(cdl_west,  "Alfalfa|Other Hay"), "hay",
+                                                                   if_else(str_detect(cdl_west,  "Rice"), "rice",
+                                                                           if_else(str_detect(cdl_west,  "Cotton"), "cotton",
+                                                                                   if_else(str_detect(cdl_west,  "Fallow"), "fallow",
+                                                                                           if_else(str_detect(cdl_west,  "Grapes"), "grapes",
+                                                                                                   if_else(str_detect(cdl_west,  "Pasture"), "pasture",
+                                                                                                           "other"))))))))))))
+types$cropgroup <- if_else(types$cdl_west=="Potatoes", "potatoes", types$cropgroup)
+types$cropgroup <- if_else(types$cdl_west=="Corn", "corn", types$cropgroup)
+types$cropgroup <- if_else(types$cdl_west=="Open Water", "Open Water", types$cropgroup)
+
+
+# Join types to the land use data frame
+df_clip2_types = left_join(df_clip2, types)
+# Plot crop categories (types)
+ggplot() + geom_raster(data=df_clip2_types, aes(x = x, y = y, fill = cropgroup)) + scale_fill_manual(values = rainbow(12))
+# Create table of cropgroup counts to calculate percentages to each type
+cropgroup.count<- df_clip2_types  %>%
+  group_by(cropgroup)  %>%
+  count
 
-ggplot(df_clip2) + geom_raster(aes(x = x, y = y, fill = updateNamesCDL(cdl_west)))
+#Checking to make sure I got all categories
+#v.count<- df_crdcount_long  %>%
+#  group_by(v)  %>%
+#  count
+#write.csv(v.count,"./v-list.csv", row.names = FALSE)

```

Comment:
Let's keep outputs outside of the top level in the /output/ sub-directory to keep things tidy. (Will be updating to the `here` infrastructure to clean up more soon too!)
```suggestion
write.csv(v.count, "./output/v-list.csv", row.names = FALSE)
```